test/odbc/psqlodbc/test_pg_varchar.cpp

```diff
@@ -0,0 +1,655 @@
+  };
+
+  const vector<string> expected_20 = {
+    STRING_20
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20, tableConstraints);
+  testPrimaryKeys(ServerType::PSQL, SCHEMA_NAME, PKTABLE_NAME, PK_COLUMNS_20);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_20, expected_20);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Varchar, Table_Unique_Constraint) {
+
+  const vector<pair<string, string>> TABLE_COLUMNS = {
+    {COL1_NAME, "INT"},
+    {COL2_NAME, DATATYPE}
+  };
+
+  const vector<string> UNIQUE_COLUMNS = {
+    COL2_NAME
+  };
+
+  string tableConstraints = createTableConstraint("UNIQUE", UNIQUE_COLUMNS);
+
+  // Insert valid values into the table and assert affected rows
+  const vector<string> inserted_values = {
+    STRING_1,
+    "" // blank value
+  };
+
+  const vector<string> expected = {
+    STRING_1,
+    ""
+  };
+
+  // table name without the schema
+  const string tableName = TABLE_NAME.substr(TABLE_NAME.find('.') + 1, TABLE_NAME.length());
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS, tableConstraints);
+  testUniqueConstraint(ServerType::PSQL, tableName, UNIQUE_COLUMNS);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values, expected);
+  testInsertionFailure(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values, false, inserted_values.size(), false);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<pair<string, string>> TABLE_COLUMNS_20 = {
+    {COL1_NAME, "INT"},
+    {COL4_NAME, DATATYPE}
+  };
+
+  const vector<string> UNIQUE_COLUMNS_20 = {
+    COL4_NAME
+  };
+
+  tableConstraints = createTableConstraint("UNIQUE", UNIQUE_COLUMNS_20);
+
+  // Insert valid values into the table and assert affected rows
+  const vector<string> inserted_values_20 = {
+    STRING_20,
+    "" // blank value
+  };
+
+  const vector<string> expected_20 = {
+    STRING_20,
+    ""
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20, tableConstraints);
+  testUniqueConstraint(ServerType::PSQL, tableName, UNIQUE_COLUMNS_20);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_20, expected_20);
+  testInsertionFailure(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_20, false, inserted_values_20.size(), false);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+  
+}
+
+TEST_F(PSQL_DataTypes_Varchar, Comparison_Functions) {
+  const vector<string> INSERTED_DATA = {
+    "One Two",
+    "Three Four"
+  };
+  const vector<string> INSERTED_PK = {
+    "1",
+    "2"
+  };
+
+  const int NUM_OF_DATA = INSERTED_DATA.size();
+  
+  // insertString initialization
+  string insertString{};
+  string comma{};
+  for (int i = 0; i < NUM_OF_DATA; i++) {
+    insertString += comma + "(" + INSERTED_PK[i] + ",\'" + INSERTED_DATA[i] + "\')";
+    comma = ",";
+  }
+
+  const vector<string> OPERATIONS_QUERY = {
+    "lower(" + COL4_NAME + ")",
```

Comment:
Can we also test `upper`, `concat`, & `trim`?

Reply:
|| THIS IS CONCAT